src/main/php/util/data/Sequence.class.php

```diff
@@ -42,18 +42,21 @@ protected function __construct($elements) {
    *
    * @param  function(): var $operation
    * @return var
+   * @throws util.data.CannotReset
    */
   protected function terminal($operation) {
     static $message= 'Underlying value is streamed and cannot be processed more than once';
```

Comment:
Is it a safe assumption that "is streamed" always holds true? The catched exception might have other causes.